src/3d/qgscameracontroller.h

```diff
@@ -48,7 +48,8 @@ class QgsCameraPose;
 class QgsVector3D;
 class QgsWindow3DEngine;
 class Qgs3DMapScene;
-class QgsCameraController4Test;
+class TestQgs3DRendering;
```

Comment:
Thanks! You can safely drop these forward declarations, they aren't required for friend classes .

Reply:
Compilation fails when I drop these forward declarations. Am I missing something?

Reply:
My bad, I was using `friend TestQgs3DRendering;` instead of `friend class TestQgs3DRendering;`. This is now fixed.